components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApplicationsApiServiceImpl.java

```diff
@@ -15,29 +25,132 @@
 
 public class ApplicationsApiServiceImpl extends ApplicationsApiService {
     @Override
-    public Response applicationsGet(String limit,String offset,String accept,String ifNoneMatch){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response applicationsGet(String subscriber, String groupId, String limit, String offset, String accept,
+            String ifNoneMatch) {
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (groupId == null) {
+            groupId = "";
+        }
+        if (subscriber == null) { //todo: implement permission
+            subscriber = username;
+        }
+
+        List<ApplicationDTO> applicationDTOList = new ArrayList<>();
+        try {
+            APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
+            Application[] applications = apiConsumer.getApplications(new Subscriber(subscriber), groupId);
+            for (Application application : applications) {
+                applicationDTOList.add(ApplicationMappingUtil.fromApplicationtoDTO(application));
+            }
+            return Response.ok().entity(applicationDTOList).build();
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
+        }
     }
+
     @Override
-    public Response applicationsPost(ApplicationDTO body,String contentType){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response applicationsPost(ApplicationDTO body, String subscriber, String contentType) {
+        //todo: validation, need to be moved
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isEmpty(body.getName())) {
+            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Name is empty"))
+                    .build();
+        }
+        if (StringUtils.isEmpty(body.getThrottlingTier())) {
+            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Tier is empty"))
+                    .build();
+        }
+        if (subscriber == null) { //todo: implement permission
+            subscriber = username;
+        }
+
+        try {
+            APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
+            Application application = ApplicationMappingUtil.fromDTOtoApplication(body, new Subscriber(subscriber));
+            String status = apiConsumer.addApplication(application, subscriber);  //todo: use "status" properly
+            return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, status)).build();
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
```

Comment:
Fixed in #1303 (https://github.com/wso2/carbon-apimgt/commit/10b904553097e950f881465d125ef962ac32a3a3)